ambassador/ambassador.py

```diff
@@ -295,18 +295,37 @@ def ambassador_stats():
 
 def new_config(envoy_base_config=None, envoy_tls_config=None,
                envoy_config_path=None, envoy_restarter_pid=None):
+    # logging.debug("new_config entry...")
+    
     rc = fetch_all_mappings()
     if not rc:
         # Failed to fetch mappings from DB.
+        logging.debug("new_config could not fetch mappings: %s" % rc)
         return rc
 
     # Suppose the fetch "succeeded" but we got no mappings?
-    if (not 'mappings' in rc) or (len(rc.mappings) <= 0):
-        # This shouldn't really happen. Weird.
-        return RichStatus.fromError("no mappings found? original rc %s" % str(rc))
+    if 'mappings' not in rc:
+        # This shouldn't happen. Clear out app.current_mappings so that
+        # we have a better chance of recovering when we get data, but don't
+        # actually change our Envoy config.
+
+        app.current_mappings = None
+
+        logging.debug("new_config got no mappings at all? %s" % str(rc))
+        return RichStatus.fromError("no mappings found at all, original rc %s" % str(rc))
+
+    num_mappings = 0
+
+    try:
+        num_mappings = len(rc.mappings)
+    except:
```

Comment:
I thought about that, and decided to deliberately go broad. 
Open to persuasion about why that would be a bad idea.